### PR TITLE
GEM-49 | Add creating groups

### PR DESCRIPTION
**Pull request by lawos98**

### CODE REVIEW

File: src/integration/kotlin/pl/edu/agh/gem/integration/client/CurrencyManagerClientIT.kt [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import pl.edu.agh.gem.internal.client.CurrencyManagerClient
+import pl.edu.agh.gem.internal.client.CurrencyManagerClientException
+import pl.edu.agh.gem.internal.client.RetryableCurrencyManagerClientException
+import pl.edu.agh.gem.util.createAvailableCurrenciesResponse
+import pl.edu.agh.gem.util.createCurrencyResponse
+
+class CurrencyManagerClientIT(
+    private val currencyManagerClient: CurrencyManagerClient,
+) : BaseIntegrationSpec({
+    should("get group members") {
```
Comment (pwgacek) [inline src/integration/kotlin/pl/edu/agh/gem/integration/client/CurrencyManagerClientIT.kt]:
```suggestion
    should("get available currencies") {
```

File: src/main/kotlin/pl/edu/agh/gem/external/persistence/GroupEntity.kt [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+private fun Member.toEntity() = MemberEntity(
+    userId = userId,
+)
+
+private fun Currency.toEntity() = CurrencyEntity(
+    code = code,
+)
+
+fun GroupEntity.toDomain() = Group(
```
Comment (pwgacek) [inline src/main/kotlin/pl/edu/agh/gem/external/persistence/GroupEntity.kt:53]: shouldn't this function be a regular function (not extension) if we declare it GroupEntity.kt?

File: src/main/kotlin/pl/edu/agh/gem/internal/model/Group.kt [hunk 3]
```diff
@@ -0,0 +1,25 @@
+package pl.edu.agh.gem.internal.model
+
+data class Group(
+    val id: String,
+    val name: String,
+    val color: Long,
+    val ownerId: String,
+    val members: Members,
+    val acceptRequired: Boolean,
+    val baseCurrency: Currencies,
```
Comment (pwgacek) [inline src/main/kotlin/pl/edu/agh/gem/internal/model/Group.kt]: according to out last talk it should be renamed to groupCurrency or just currency.

**Review by pwgacek — CHANGES_REQUESTED** [review]

File: src/main/kotlin/pl/edu/agh/gem/external/persistence/GroupEntity.kt (hunk 2 above)
Comment (lawos98) [reply src/main/kotlin/pl/edu/agh/gem/external/persistence/GroupEntity.kt:53]: It's an opinionated style-oriented rule to not have functions inside data classes

Comment (lawos98) [reply src/main/kotlin/pl/edu/agh/gem/external/persistence/GroupEntity.kt:53]: But there's nothing inherently wrong with having methods in your data classes

**Review by pwgacek — APPROVED** [review]

**Pull request merged**